### ISSUE TITLE
The autocomplete popup should not appear on typing a left square bracket if no specific suggestions are available

### ISSUE THREAD

**Issue by atrigent** [microsoft/pylance-release#3224]
Yes, I know this is similar to the pre-existing issue #1712 and the discussion #2599 that was created from it. However, I'm not asking for the popup to only contain relevant suggestions or those that would be syntactically valid - I'm suggesting that the popup should not appear at all in this context. I believe it does because, In some cases when typing a `[`, such as when indexing a dictionary that was defined nearby or for which type information is available, Pylance is able to make specific, useful suggestions. However, in most other situations it can't. Pylance doesn't open the autocomplete popup in any other situations where no particular information or specific suggestions are available, because It is more reasonable then to wait until the user starts typing a name or explicitly invokes the autocomplete dialog. Pylance should also do this in such cases when `[` is typed.

<h3> Environment data </h3>

-   Language Server version: 2022.8.30
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.8
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: off

<h3> Repro Steps </h3>

1. Type a `[` in an editor where pylance is activated in a situation where no specific suggestions are available
2. See a not particularly useful autocomplete popup appear

<h3> Expected behavior </h3>

This popup should not appear, just as it doesn't in any other analogous situations.

<h3> Actual behavior </h3>

A popup appears with some generic suggestions.

**Comment by debonte** [comment]
I'm guessing that the user is seeing the keyword and stdlib suggestions, like this:
![image](https://user-images.githubusercontent.com/12820898/185973435-d5478863-082a-4ab8-a96b-db369c57617d.png)

@atrigent, can you confirm? If you're seeing something different, please provide a code sample and a screenshot of the completion list you are getting.

**Comment by atrigent** [comment]
Yes, that is what I'm seeing. It's the same thing the person was seeing in #1712.
